### PR TITLE
add adelais PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Adelais en ciel - compatible with TGR Alice and all clone variants

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/adelais/keymaps/default/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Team Mechlovin
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/mechlovin/adelais/rules.mk [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rules.mk:14]:
```suggestion
```

**Comment by fauxpark** [comment]
What's the difference between this and #8944?

**Comment by mechlovin** [comment]
> What's the difference between this and #8944?

**Comment by mechlovin** [comment]
this is standard backlighting, #8944 is rgb pcb

**Comment by mechlovin** [comment]
> this is standard backlighting (i mean soldering LED), #8944 is rgb pcb

**Comment by mechlovin** [comment]
could i merge them?

**Comment by fauxpark** [comment]
So is it the exact same PCB?

If so, adding two keyboards isn't the way to do this. You can just enable both features in the same config. Since this is using an ARM chip, there's plenty of flash space to accommodate it.

**Comment by mechlovin** [comment]
Oh, got it. Thank you.
May i close this PR, #8944 and open new PR?

**Comment by fauxpark** [comment]
You don't need to open a new PR. Just use one of these two and close the other.

**Comment by mechlovin** [comment]
okie, thank you!

**Comment by fauxpark** [comment]
The config.h and rules.mk stuff can be deduped here, you don't need separate keymaps for the features.

**Comment by fauxpark** [comment]
Oh, I see, there are two versions of this PCB after all, one with RGB matrix, and one with underglow + backlight. So, instead of separate keymaps you should make subfolders inside the `adelais` folder for each version with just the relevant config/rules stuff. Anything that's common to both can be in the top-level rules.mk/config.h.

**Comment by mechlovin** [comment]
Exactly, these are the 2 PCB versions. But they have the same layout, matrix and pinout.
So I think what I changed is fine :D

**Comment by mechlovin** [comment]
Hi @fauxpark, Do I have to change anything else?

**Comment by fauxpark** [comment]
As I said, don't use separate keymaps for this. Make subfolders with the different config options, and leave stuff common to both in the top-level config/rules.

**Comment by mechlovin** [comment]
I just updated according to what I understand.
Am I doing it right?
Thank you! :D

File: keyboards/mechlovin/adelais/config.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x4D4C // ML-Mechlovin
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/config.h:23]:
```suggestion
#define VENDOR_ID    0x4D4C // "ML"
#define DEVICE_VER   0x0001
#define MANUFACTURER Team.Mechlovin
```

File: keyboards/mechlovin/adelais/rgb_led/config.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#define PRODUCT_ID   0xAEC1
+#define DEVICE_VER   0x0001
+#define MANUFACTURER Team.Mechlovin
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rgb_led/config.h:5]:
```suggestion
```

File: keyboards/mechlovin/adelais/standard_led/config.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#define PRODUCT_ID   0xAD01
+#define DEVICE_VER   0x0001
+#define MANUFACTURER Team.Mechlovin
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/standard_led/config.h:5]:
```suggestion
```

File: keyboards/mechlovin/adelais/standard_led/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# adelais
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/standard_led/readme.md:1]: This file can be deleted.

File: keyboards/mechlovin/adelais/rgb_led/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# adelais_en_ciel
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/rgb_led/readme.md:1]: This file can be deleted.

File: keyboards/mechlovin/adelais/readme.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# adelais
+
+![adelais](https://i.imgur.com/6U1IfZe.png)
+
+compatible with TGR Alice and all clone variants. Comes with RGB underglow pre-soldered.
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/readme.md:5]:
```suggestion
Compatible with TGR Alice and all clone variants.  
`standard_led` (Adelais) comes with backlight and RGB underglow LEDs pre-soldered.  
`rgb_led` (Adelais En Ciel) comes with per-key RGB LEDs pre-soldered.
```

File: keyboards/mechlovin/adelais/readme.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# adelais
```
Comment (fauxpark) [inline keyboards/mechlovin/adelais/readme.md:1]:
```suggestion
# Adelais / Adelais En Ciel
```

**Comment by mechlovin** [comment]
Awesome, thank you!

**Comment by mechlovin** [comment]
Hi @drashna. Could you please help me to check them again?

File: keyboards/mechlovin/adelais/info.json [hunk 10]
```diff
@@ -0,0 +1,80 @@
+{   "keyboard_name": "Adelais",
+    "url": "",
+    "maintainer": "Team Mechlovin",
+    "width": 18,
+    "height": 6,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 66,
+            "layout": [
```
Comment (noroadsleft) [inline keyboards/mechlovin/adelais/info.json:9]: This tree is a mess, but it'll be easier for me to debug if this gets merged as is.

image

Comment (mechlovin) [reply keyboards/mechlovin/adelais/info.json:9]: I have just discovered, my layout is similar to the @Maartenwut  pcb layout.
Hello @Maartenwut. Can I use the info.json file? I'll appreciate that. :D

Comment (noroadsleft) [reply keyboards/mechlovin/adelais/info.json:10]: Resolved.

image

**Pull request merged**